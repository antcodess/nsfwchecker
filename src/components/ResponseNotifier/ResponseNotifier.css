.response-section {
  margin-top: 15px;
  margin-bottom: 60px;
}

.response-text {
  font-size: 1.5rem;
  margin-top: 20px;
}

.response-new-search {
  margin-top: 50px;
  text-align: center;
}

.response-new-search p {
  font-weight: bold;
  margin-bottom: 5px;
}

/* .response-new-search p:hover {
  cursor: pointer;
  color: #d3d3d3;
} */

.response-new-search a {
  text-decoration: none;
  color: #808080;
}

.response-new-search a:hover {
  text-decoration: none;
  color: #d3d3d3;
}

.new-search-button {
  background: #ff0000;
  color: white;
  font-weight: bold;
  border: none;
  border-radius: 50px;
  padding: 20px;
  cursor: pointer;
  outline: none;
}

.new-search-button:hover {
  background: black;
}


