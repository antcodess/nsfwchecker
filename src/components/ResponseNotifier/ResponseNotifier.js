import React from 'react';
import alertIcon from './alert-icon-red.png';
import approvedIcon from './approved-icon-green.png';
import errorIcon from './error-icon-yellow.png';
import './ResponseNotifier.css';

const ResponseNotifier = ({ answer, description, onNewSearch, imageUrl }) => {
  
  let answerIcon;
  if (answer === 'nsfw') {
    answerIcon = alertIcon;
  } else if (answer === 'sfw') {
    answerIcon = approvedIcon;
  } else {
    answerIcon = errorIcon;
  }

  return (
    <section className="response-section">
      <div>
        <img alt='' src={answerIcon} width='80' height='auto' />
      </div>
      <div className="response-text">
        <p>{description}</p>
      </div>
      {answer === 'sfw' ? 
        <div className="response-new-search">
          <p>url:</p><a href={imageUrl} target='_blank' rel="noopener noreferrer">{imageUrl}</a>
        </div> 
      : <div>
        </div> }
      <div className="response-new-search">
        <button className="new-search-button" onClick={onNewSearch}>New Search</button>
      </div>
    </section>
  )
}

export default ResponseNotifier