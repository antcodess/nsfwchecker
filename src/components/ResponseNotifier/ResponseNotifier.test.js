import React from 'react';
import { shallow } from 'enzyme';
import ResponseNotifier from './ResponseNotifier';

it('expects to render ResponseNotifier component', () => {
  expect(shallow(<ResponseNotifier />)).toMatchSnapshot();
});