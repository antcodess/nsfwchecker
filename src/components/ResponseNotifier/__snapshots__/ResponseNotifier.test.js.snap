// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render ResponseNotifier component 1`] = `ShallowWrapper {}`;
